pkg/cli/login/loginoptions.go

```diff
@@ -78,12 +80,17 @@ func NewLoginOptions(streams genericclioptions.IOStreams) *LoginOptions {
 // Gather all required information in a comprehensive order.
 func (o *LoginOptions) GatherInfo() error {
 	if err := o.gatherAuthInfo(); err != nil {
+		// Display legal notice even if authentication failed
+		o.prepareAndDisplayLegalNotice()
```

Comment:
If auth failed, how will you be able to get anything from the server?

Reply:
That would be handled by RBAC. This is the role I created for it:

```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: legalnotice-reader
  namespace: openshift-config
rules:
- apiGroups: [""]
  resources: ["configmaps"]
  resourceNames: ["legal-notice"]
  verbs: ["get"]
```

And this is the role binding:

```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: read-legalnotice-global
subjects:
- kind: Group
  name: system:authenticated
  apiGroup: rbac.authorization.k8s.io
- kind: Group
  name: system:unauthenticated
  apiGroup: rbac.authorization.k8s.io
roleRef:
  kind: ClusterRole
  name: legalnotice-reader
  apiGroup: rbac.authorization.k8s.io
```

As you can see, all users (authenticated and not authenticated) can read that message. Which is necessary for legal reasons.